odl/solvers/functional/functional.py

```diff
@@ -822,19 +822,22 @@ def __str__(self):
                                           self.translation)
 
 
-class FunctionalLinearPerturb(Functional):
+class FunctionalQuadraticPerturb(Functional):
 
-    """The ``Functional`` representing ``f(.) + <linear_term, .>``."""
+    """The ``Functional`` representing ``F(.) + a * |.|^2 + <., u>``."""
 
-    def __init__(self, func, linear_term):
+    def __init__(self, func, quadratic_term=0, linear_term=None):
         """Initialize a new instance.
 
         Parameters
         ----------
         func : `Functional`
             Function corresponding to ``f``.
-        linear_term : `domain` element
+        quadratic_term : ``domain.field`` element, optional
```

Comment:
In some sense, the `linear_term` (name also not ideal) is also just a vector of coefficients, at least in the discrete case. Otherwise it's more of a factor than a term.